src/app/location.service.ts

```diff
@@ -0,0 +1,44 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, catchError, map, throwError } from 'rxjs';
+
+interface LocationResponse {
+  results: Array<{ uuid: string; display: string }>;
+}
+
+type LocationOption = { value: string; label: string };
+
+@Injectable({ providedIn: 'root' })
+export class LocationService {
+  constructor(private http: HttpClient) {}
+
+  getLocations(
+    search: string,
+    options?: { tag?: string }
+  ): Observable<LocationOption[]> {
+    const apiUrl = 'https://dev3.openmrs.org/openmrs/ws/rest/v1/location';
```

Comment:
Is this for testing?

Reply:
Yes thats correct, will have to register a datasource that  is able to filter locations by tag